js/ai/src/extract.ts

```diff
@@ -56,3 +60,46 @@ export function extractJson<T = unknown>(text: string): T | null {
   }
   throw new Error(`No JSON object or array found in model output: ${text}`);
 }
+
+export function extractAndUntruncateJson<T = unknown>(text: string): T | null {
```

Comment:
I mean, I totally understand understand this by reading the code, just asking for a friend, can you explain a bit what this function does any why it's necessary? maybe a comment (with examples) would be helpful...

Reply:
I was mostly following the logic of the existing `extractJson` function above it, but maybe I should have just added an optional parseFn argument to `extractJson` instead of repeating it haha

But I can also add comments no problem

Reply:
I've updated the PR so that i'm just modifying the existing method, i've added some comments too, wdyt?